Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/service/SimulationInfoService.java

```diff
@@ -306,11 +313,37 @@ private void deriveGatewayProbabilities(
                     .map(EdgeFrequency::getFrequency)
                     .reduce(0.0D, Double::sum);
 
+                // Set the percentage for each edge's frequency
+                gatewayEntry.getValue().forEach(edgeFrequency ->
+                    edgeFrequency.setPercentage(BigDecimal.valueOf(edgeFrequency.getFrequency() / totalFrequency)
+                        .setScale(4, RoundingMode.HALF_UP).doubleValue()));
+
+                // Determine if the percentages add up to a 100%
+                double totalProbabilities = gatewayEntry.getValue().stream()
+                    .map(EdgeFrequency::getPercentage)
+                    .reduce(0.0D, Double::sum);
```

Comment:
Perhaps `https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/DoubleStream.html#sum()` instead?

Reply:
Yeah. Thanks for that. much cleaner 👍